tests/test_consumer.py

```diff
@@ -0,0 +1,169 @@
+    """
+    Test the single_day() function from the consumer module
+    Args:
+        monkeypatch: patch for necessary attributes
+    """
+
+    # Mock the decide() function from the consumers module
+    def mock_decide(*args, **kwargs):
+        return True  # force decision to always be true
+
+    # Mock the consume() function from the consumers module
+    def mock_consume(*args, **kwargs):
+        return 1  # force consumption to always be 1
+
+    monkeypatch.setattr(cs, "decide", mock_decide)
+    monkeypatch.setattr(cs, "consume", mock_consume)
+
+    customer = test_cfg.customers["test"]
+    test_output = cs.single_day(customer_config=customer, day_of_week=0)
+
+    # Positive testing
+    assert test_output == 1
+
+    # Negative testing
+    with pytest.raises(TypeError):
+        cs.single_day(
+            customer_config=customer, day_of_week="not_an_int"
+        )  # must be an int
+    with pytest.raises(ValueError):
+        cs.single_day(customer_config=customer, day_of_week=7)  # must in range 0-6
+
+
+def test_multi_day(monkeypatch):
+    """
+    Test the multi_day() function from the consumer module
+    Args:
+        monkeypatch: patch for necessary attributes
+    """
+
+    # Mock the get_customer() function from consumers module
+    def mock_get_customer(*args, **kwargs):
+        return test_cfg.customers["test"]
+
+    # Mock the single_day() function from consumers module
+    def mock_single_day(*args, **kwargs):
+        return 1  # force consumption to always be 1
+
+    monkeypatch.setattr(cs, "get_customer", mock_get_customer)
+    monkeypatch.setattr(cs, "single_day", mock_single_day)
+
+    test_output = cs.multi_day(customer_number="0", days=2)
+
+    # Positive testing
+    assert test_output["servings"].iloc[0] == 1
+    assert test_output["servings"].iloc[1] == 1
+
+    # Negative testing
+    with pytest.raises(TypeError):
+        assert cs.multi_day(customer_number=0, days=1)  # must be a string
+    with pytest.raises(TypeError):
+        assert cs.multi_day(customer_number="0", days="not_int")  # must be an int
+
+
+def test_get_customer(monkeypatch):
+    """
+    Test the get_customer() function from the consumer module
+    Args:
+        monkeypatch: patch for necessary attributes
+    """
+
+    # Mock the customers configurations in the customers module
+    def mock_customers():
+        return test_cfg.customers
+
+    monkeypatch.setattr(ct, "customers", mock_customers())
+
+    # Positive testing
+    valid_output = test_cfg.customers["test"]
+    test_output = cs.get_customer(customer_number="test")
+    assert valid_output == test_output
+
+    # Negative testing
+    with pytest.raises(Exception):
+        assert cs.get_customer(customer_number="not_a_customer")
+
+
+def test_write_output():
+    """
+    Test the write_output() function of the consumer module
+    """
+    from tests import test_components
+
+    test_path = pathlib.Path.cwd() / "test_components"
```

Comment:
This path changes based on whether you run the test from the project root (which is what Travis is doing) or from the inside the `tests` folder. I think that's whats causing your tests to fail on Travis. Alternative (I think) would be `pathlib.Path(__file__)` which would return the path of the *script* that's running, not the current working directory.

But, if you're creating and reading files during your tests using this is a better option: https://docs.pytest.org/en/stable/tmpdir.html#temporary-directories-and-files
That also has the benefit of cleaning up after the tests run.